### ISSUE TITLE
Broken widgets for suggestions

### ISSUE THREAD

**Issue by bradfordcondon** [statonlab/tripal_hq#16]
```
Notice: Undefined index: chado-analysis_relationship__type_id in sbo__relationship_widget->validate() (line 368 of /Users/chet/UTK/tripal/sites/all/modules/tripal/tripal_chado/includes/TripalFields/sbo__relationship/sbo__relationship_widget.inc).

```
It's not just the relationship wiget though.  Every field with cardinality appears to be create a new null record upon submission.  I'm guessing this is because of the wonky way fields handle submissions--- ie we serialize a null value when the actual submission would unset the field.

see below image aftr i've submitted a bunch of edits.

![screen shot 2018-10-09 at 4 03 48 pm](https://user-images.githubusercontent.com/7063154/46695223-f183f200-cbdc-11e8-99a5-be4b44ee0bd3.png)

**Comment by almasaeed2010** [comment]
All of those fields, oddly enough, have a value of `delete_me`:

<img width="855" alt="screen shot 2018-10-15 at 10 54 59 am" src="https://user-images.githubusercontent.com/1512664/46958869-c8de7b00-d068-11e8-90f7-188ad1d71029.png">


Are we supposed to loop through and delete these fields? How does tripal handle this and when does it handle it?

**Comment by bradfordcondon** [comment]
ugh.  it's because if there is no value, nothing happens.  So if t was previously set, and you nulled it, it wouldnt delete.  This way, theres  a value so that the field storage happens, and it sees the empty cvterm_id and analysis_id stuff and deletes the record.

**Comment by almasaeed2010** [comment]
Fixed in #39
